src/main/java/com/github/underscore/U.java

```diff
@@ -446,13 +447,7 @@ public void forEachRight(final Consumer<? super T> func) {
     }
 
     public List<T> filter(final Predicate<T> pred) {
-        final List<T> filtered = newArrayList();
-        for (final T element : value()) {
-            if (pred.test(element)) {
-                filtered.add(element);
-            }
-        }
-        return filtered;
+        return newArrayList(filter(value(), pred));
```

Comment:
It seems that `newArrayList()` is called twice.

Reply:
I don't see it, where is that? Which rows?